services/ui-src/src/components/cards/EntityCard.tsx

```diff
@@ -45,33 +41,81 @@ export const EntityCard = ({
-          {data.category}
+          {formattedEntityData.category}
         </Heading>
-        <Text sx={sx.description}>{data.standardDescription}</Text>
+        <Text sx={sx.description}>
+          {formattedEntityData.standardDescription}
+        </Text>
         <Text sx={sx.subtitle}>General category</Text>
-        <Text sx={sx.subtext}>{data.standardType}</Text>
+        <Text sx={sx.subtext}>{formattedEntityData.standardType}</Text>
         <Button
           variant="outline"
           size="sm"
-          sx={sx.editEntityButton}
+          sx={sx.editButton}
+          data-testid="editMeasureButton"
           leftIcon={<Image src={editIcon} alt="edit icon" height="1rem" />}
         >
           Edit measure
         </Button>
-        <Text sx={sx.unfinishedMessage}>
-          Complete the remaining indicators for this access measure by entering
-          details.
-        </Text>
-        <Button size="sm" sx={sx.enterDrawerButton}>
-          Enter details
-        </Button>
+        {entityCompleted ? (
+          <>
+            <Flex sx={sx.highlightContainer}>
+              <Box sx={sx.highlightSection}>
+                <Text sx={sx.subtitle}>Provider</Text>
+                <Text sx={sx.subtext}>{formattedEntityData?.provider}</Text>
+              </Box>
+              <Box sx={sx.highlightSection}>
+                <Text sx={sx.subtitle}>Region</Text>
+                <Text sx={sx.subtext}>{formattedEntityData?.region}</Text>
+              </Box>
+              <Box sx={sx.highlightSection}>
+                <Text sx={sx.subtitle}>Population</Text>
+                <Text sx={sx.subtext}>{formattedEntityData?.population}</Text>
+              </Box>
+            </Flex>
+            <Text sx={sx.subtitle}>Monitoring Methods</Text>
+            <Text sx={sx.subtext}>
+              {formattedEntityData?.monitoringMethods.join(", ")}
+            </Text>
+            <Text sx={sx.subtitle}>Frequency of oversight methods</Text>
+            <Text sx={sx.subtext}>{formattedEntityData.methodFrequency}</Text>
+          </>
+        ) : (
+          <Text sx={sx.unfinishedMessage}>
+            Complete the remaining indicators for this access measure by
+            entering details.
+          </Text>
+        )}
+        {entityCompleted ? (
+          <Button
+            variant="outline"
+            size="sm"
+            sx={sx.editButton}
+            data-testid="editDetailsButton"
+            leftIcon={<Image src={editIcon} alt="edit icon" height="1rem" />}
+            onClick={() => openDrawer(entity)}
+          >
+            Edit Details
+          </Button>
+        ) : (
+          <Button
+            size="sm"
+            sx={sx.openDrawerButton}
+            data-testid="enterDetailsButton"
+            onClick={() => openDrawer(entity)}
+          >
+            Enter details
+          </Button>
+        )}
       </Box>
     </Card>
   );
 };
 
 interface Props {
   entity: AnyObject;
```

Comment:
We probably want to change this type to `EntityShape` to match the other PRs.
```suggestion
  entity: EntityShape;
```